hamilton/plugins/sklearn_plot_extensions.py

```diff
@@ -88,8 +89,12 @@ def _get_saving_kwargs(self) -> Dict[str, Any]:
         return kwargs
 
     def save_data(self, data: SKLEARN_PLOT_TYPES_ANNOTATION) -> Dict[str, Any]:
-        data.plot()
-        data.figure_.savefig(self.path, **self._get_saving_kwargs())
+        if isinstance(data, pyplot.Figure):
+            figure = data
+        else:
+            data.plot()
+            figure = data.figure_
+        figure.savefig(self.path, **self._get_saving_kwargs())

```

Comment:
does this need to be rebased?

Reply:
No this needs to be broken into a separate commit, nice noticing